macros/PGbasicmacros.pl

```diff
@@ -2908,12 +2908,7 @@ sub row {
 =head2 Macros for displaying static images
 
     Usage:
-    $string = image($image, width => 100, height => 100, tex_size => 800)
-    $string = image($image, width => 100, height => 100, extra_html_tags => 'align="middle"', tex_size => 800)
-    $string = image([$image1, $image2], width => 100, height => 100, tex_size => 800)
```

Comment:
I don't think that you should remove the array ref version of this.  It is still supported.

Reply:
The thing is that because of how `wantarray` is used in the return statement, it can't easily be used that way. It is wired so that if you use `\{ image([$gr1,$gr2]) \}` you will only see `$gr1`. It's being called in scalar context, and its written to only return the first image in that situation. Not sure why, maybe something to do with the legacy code.

So you'd have to use it in some way to account for this. It works to do something like:
`\{ join(' ',image([$gr1,$gr2])) \}`

Useful if you understand about arrays, about image()'s ability to return an array or a scalar, and if you understand join.

I'll leave it there, maybe with more explanation though.

Reply:
So build the `join` into the image return.  In other words change
`return wantarray ? @output_list : $output_list[0];`
to
`return wantarray ? @output_list : join(' ', @output_list);`

Reply:
One idea: in scalar context, it could go ahead and join all the images with a default space character (or something) rather than only return the first image.

Reply:
Heh, posted to a page that had not refreshed with your comment.

Reply:
👍